code/lib/builder-vite/src/utils/without-vite-plugins.ts

```diff
@@ -1,16 +1,21 @@
 import type { PluginOption } from 'vite';
 
 // recursively remove all plugins with the given names
-export const withoutVitePlugins = (
+export const withoutVitePlugins = async (
   plugins: PluginOption[] = [],
   namesToRemove: string[]
-): PluginOption[] =>
-  plugins.map((plugin) => {
+): Promise<PluginOption[]> => {
+  const result = [];
+  const resolvedPlugins = await Promise.all(plugins);
+  // eslint-disable-next-line no-restricted-syntax -- we need to await in the loop
+  for (const plugin of resolvedPlugins) {
     if (Array.isArray(plugin)) {
-      return withoutVitePlugins(plugin, namesToRemove);
+      // eslint-disable-next-line no-await-in-loop
+      result.push(await withoutVitePlugins(plugin, namesToRemove));
     }
-    if (plugin && 'name' in plugin && namesToRemove.includes(plugin.name)) {
-      return false;
+    if (plugin && 'name' in plugin && !namesToRemove.includes(plugin.name)) {
+      result.push(plugin);
     }
-    return plugin;
-  });
+  }
+  return result;
+};
```

Comment:
This codebase has an aversion against for loops, but due to the nested asynchronousness I couldn't figure out how to make it work with regular Array functions.

Open to suggestions.